### PR TITLE
ZFS-1459: Delete the driver package from the Driver Store during uninstall.

### PR DESCRIPTION
**Pull request by datacore-rm**
Jira:
ZFS-1459: Delete the driver package from the Driver Store during uninstall.

Issue: 
1) During the ZFS driver uninstall, the driver package is not removed from the driver store.
2) The Driver file is not deleted from the 'system32\drivers" directory.
3) Install and uninstall test signed version of the driver. Then install the HLK signed version of the same driver file. Windows does not allow to overwrite as the actual file version in the zfsin.inf file is same. 
 
Reproduce:-
1) After uninstall, running the following cmd lists the driver oem file from "C:\Windows\INF" folder.
_PS C:\Users\Administrator> $driver = get-windowsdriver -online -all | ? {($_.OriginalFileName -match "zfsin.inf")}
PS C:\Users\Administrator> $driver.Driver_

2) Enumerate the driver store packages using 
`pnputil /enum-drivers`

Fix:
As per msdn documentation, the installer has to call SetupUninstallOEMInf() function.

https://learn.microsoft.com/en-us/windows-hardware/drivers/install/using-setupapi-to-uninstall-devices-and-driver-packages#-deleting-a-driver-package-from-the-driver-store.

Test by Dev:
`C:\ZFSin>zfsinstaller.exe uninstall .\ZFSin.inf
DefaultUninstall 128 .\ZFSin.inf
uninstallRootDevice:
uninstallRootDevice: looking for device rootnode to remove...
uninstallRootDevice: device found, removing ...
uninstallRootDevice: exit 0:0x0
Executing unlodctr /m:"C:\ZFSin\.\OpenZFS.man"

Info: Successfully uninstalled the performance counters from the counter definition XML file C:\ZFSin\.\OpenZFS.man.
**Successfully deleted the driver package from Driver Store.oemfile:oem12.inf
Successfully deleted the driver file:C:\Windows\system32\drivers\ZFSin.sys**
`

### CODE REVIEW

File: cmd/os/windows/zfsinstaller/zfsinstaller.cpp [hunk 1]
```diff
@@ -895,3 +898,102 @@ uninstallRootDevice(char *inf)
...
+
+		if (0 < GetPrivateProfileString(
+			_T("SourceDisksFiles"),	// section name
+			szDriverFileName,		// lpKeyName
+			NULL,
+			szValue,				// lpReturnedString
+			MAX_PATH,
+			oemName.c_str()))		// lpFileName
+		{
+			BOOL success = SetupUninstallOEMInf(findFileData.cFileName, SUOI_FORCEDELETE, NULL);
```
Comment (arun-kv) [inline cmd/os/windows/zfsinstaller/zfsinstaller.cpp:955]: Do we need to explicitly remove the .inf and .pnf file or SetupUninstallOEMInf will take care of it?

File: cmd/os/windows/zfsinstaller/zfsinstaller.cpp [hunk 2]
```diff
@@ -895,3 +898,102 @@ uninstallRootDevice(char *inf)
...
+			if (!success)
+			{
+				error = GetLastError();
+				std::wcout << "\nFailed to delete the driver package from Driver Store.oemfile:" << findFileData.cFileName << ",error:" << error << std::endl;
+			}
+			else
+				std::wcout << "\nSuccessfully deleted the driver package from Driver Store.oemfile:" << findFileData.cFileName << std::endl;
```
Comment (arun-kv) [inline cmd/os/windows/zfsinstaller/zfsinstaller.cpp:962]: Why do we need to continue search after success? Is it possible to have multiple INF file for a driver?

File: cmd/os/windows/zfsinstaller/zfsinstaller.cpp (hunk 1 above)
Comment (datacore-rm) [reply cmd/os/windows/zfsinstaller/zfsinstaller.cpp:955]: As documented in SetupUninstallOEMInf(), it deletes the files as well.
`The SetupUninstallOEMInf function uninstalls a specified .inf file and any associated .pnf file.`

File: cmd/os/windows/zfsinstaller/zfsinstaller.cpp (hunk 2 above)
Comment (datacore-rm) [reply cmd/os/windows/zfsinstaller/zfsinstaller.cpp:962]: In the case of upgrade installations, there would be multiple packages installed. So, we need to cleanup all of them.

_PS C:\Users\Administrator> $driver = get-windowsdriver -online -all | ? {($_.OriginalFileName -match "zfsin.inf")}
PS C:\Users\Administrator> $driver.Driver
oem12.inf
oem13.inf

**Review by arun-kv — APPROVED** [review]

**Comment by datacore-rm** [comment]
Thanks @arun-kv.

**Pull request merged**